llm-chain-openai/examples/similarity_search.rs

```diff
@@ -0,0 +1,86 @@
+use std::sync::Arc;
+
+use llm_chain::{schema::Document, traits::VectorStore, vectorstores::qdrant::Qdrant};
+use llm_chain_openai::embeddings::Embeddings;
+use qdrant_client::{
+    prelude::{QdrantClient, QdrantClientConfig},
+    qdrant::{CreateCollection, Distance, VectorParams, VectorsConfig},
+};
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() {
+    // Qdrant prep
+    let config = QdrantClientConfig::from_url("http://localhost:6334");
+    let client = Arc::new(QdrantClient::new(Some(config)).await.unwrap());
+    let collection_name = "my-collection".to_string();
+    let embedding_size = 1536;
+
+    if !client
+        .has_collection(collection_name.clone())
+        .await
+        .unwrap()
+    {
+        client
+            .create_collection(&CreateCollection {
+                collection_name: collection_name.clone(),
+                vectors_config: Some(VectorsConfig {
+                    config: Some(qdrant_client::qdrant::vectors_config::Config::Params(
+                        VectorParams {
+                            size: embedding_size,
+                            distance: Distance::Cosine.into(),
+                            hnsw_config: None,
+                            quantization_config: None,
+                        },
+                    )),
+                }),
+                ..Default::default()
+            })
+            .await
+            .unwrap();
+    }
+    println!("OPENAI KEY: {}", std::env::var("OPENAI_API_KEY").unwrap());
```

Comment:
Is this to require us to have an OPENAI_API_KEY?

Reply:
This is a leftover from my debugging session, I'll remove this

This is becuase the new() constructor for async_openai::Client() silently fails reading the key from env vars and sets an empty string.

I think I'll change these two examples to use a struct literal instead of new() so the user can actually see the env read operation in the example itself and there is no confusion as to where and why this fails.

Reply:
Nvm, I think using Embeddings::default() makes the example easier to understand and the unwrap() above it is a good way to indicate that the key is required, at least for now